cmd/create/cluster/cmd.go

```diff
@@ -511,94 +520,66 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	operatorIAMRoles := args.operatorIAMRoles
-	operatorIAMRoleList := []ocm.OperatorIAMRole{}
+	operatorRolesPrefix := args.operatorRolesPrefix
 	if roleARN != "" {
-		for _, role := range operatorIAMRoles {
-			if !strings.Contains(role, ",") {
-				reporter.Errorf("Expected operator IAM roles to be a comma-separated " +
-					"list of name,namespace,role_arn")
-				os.Exit(1)
-			}
-			roleData := strings.Split(role, ",")
-			if len(roleData) != 3 {
-				reporter.Errorf("Expected operator IAM roles to be a comma-separated " +
-					"list of name,namespace,role_arn")
+		if operatorRolesPrefix == "" {
+			operatorRolesPrefix = clusterName
+		}
+		if interactive.Enabled() {
+			operatorRolesPrefix, err = interactive.GetString(interactive.Input{
+				Question: "Operator roles prefix",
+				Help:     cmd.Flags().Lookup("operator-roles-prefix").Usage,
+				Required: true,
+				Default:  operatorRolesPrefix,
```

Comment:
Should this be `clusterName` and we don't set `operatorRolesPrefix` so that the `--prefix` command works without generating a prompt?

Reply:
We already set it to cluster name above:
```
		if operatorRolesPrefix == "" {
			operatorRolesPrefix = clusterName
		}
```
And the prompt only shows up if `--interactive` is sent (or if the command is missing the `--cluster-name` flag, which is the only required one)

Reply:
Oh and the `--interactive` flag is set if you don't sent all flags?

```
rosa init account --mode auto -y --prefix james
I: Logged in as 'jaharrin.openshift' on 'https://api.stage.openshift.com'
I: Validating AWS credentials...
I: AWS credentials are valid!
I: Validating AWS quota...
I: AWS quota ok. If cluster installation fails, validate actual AWS resource usage against https://docs.openshift.com/rosa/rosa_getting_started/rosa-required-aws-service-quotas.html
I: Verifying whether OpenShift command-line tool is available...
I: Current OpenShift Client Version: 4.7.9
? OpenShift version: 4.7
? Role prefix: [? for help] (james) 
```

Reply:
That's for a different command. For `init account`/`create account-roles`, the `--prefix` and `--version` flags are required, so if you don't set the flags it drops you into interactive mode. For `create cluster`, only `--cluster-name` is required, so if you don't set that it drops you into interactive mode (but if you just do `rosa create cluster --cluster-name foo` it will just create the cluster without prompts - unless you're using a local STS session, in which case you also need `--role-arn`)

Reply:
:facepalm: its been a long day.